### ISSUE TITLE
ci - trunk failure mailer tests

### ISSUE THREAD

**Issue by kapilt** [cloud-custodian/cloud-custodian#3422]
On last trunk merge, seeing new failures wrt to mailer unit tests around redis. need to investigate new versions of fake redis, and redis client.

looks like an issue with fakeredis given its release history
https://pypi.org/project/fakeredis/#history

went ahead and filed an issue against fakeredis to see if there's additional details on compatibility constraints.

https://github.com/jamesls/fakeredis/issues/233

```
2019-01-24T11:28:53.5373154Z =================================== FAILURES ===================================
2019-01-24T11:28:53.5373831Z ________ EmailTest.test_email_to_email_message_map_without_ldap_manager ________
2019-01-24T11:28:53.5376815Z [gw0] linux2 -- Python 2.7.15 /home/vsts/work/1/s/.tox/py27-cov/bin/python
2019-01-24T11:28:53.5376956Z Traceback (most recent call last):
2019-01-24T11:28:53.5377026Z   File "/home/vsts/work/1/s/tools/c7n_mailer/tests/test_email.py", line 53, in setUp
2019-01-24T11:28:53.5377081Z     self.email_delivery = MockEmailDelivery(MAILER_CONFIG, self.aws_session, logger)
2019-01-24T11:28:53.5377156Z   File "/home/vsts/work/1/s/tools/c7n_mailer/c7n_mailer/email_delivery.py", line 78, in __init__
2019-01-24T11:28:53.5377208Z     self.ldap_lookup = self.get_ldap_connection()
2019-01-24T11:28:53.5377279Z   File "/home/vsts/work/1/s/tools/c7n_mailer/tests/test_email.py", line 46, in get_ldap_connection
2019-01-24T11:28:53.5377511Z     return get_ldap_lookup(cache_engine='redis')
2019-01-24T11:28:53.5377577Z   File "/home/vsts/work/1/s/tools/c7n_mailer/tests/common.py", line 359, in get_ldap_lookup
2019-01-24T11:28:53.5377804Z     ldap_lookup.caching.set('michael_bolton', michael_bolton)
2019-01-24T11:28:53.5377860Z   File "/home/vsts/work/1/s/tools/c7n_mailer/c7n_mailer/ldap_lookup.py", line 205, in set
2019-01-24T11:28:53.5377929Z     return self.connection.set(key, json.dumps(value))
2019-01-24T11:28:53.5378348Z   File "/home/vsts/work/1/s/.tox/py27-cov/lib/python2.7/site-packages/redis/client.py", line 1394, in set
2019-01-24T11:28:53.5378576Z     return self.execute_command('SET', *pieces)
2019-01-24T11:28:53.5378834Z   File "/home/vsts/work/1/s/.tox/py27-cov/lib/python2.7/site-packages/redis/client.py", line 754, in execute_command
2019-01-24T11:28:53.5378890Z     connection.send_command(*args)
2019-01-24T11:28:53.5379312Z   File "/home/vsts/work/1/s/.tox/py27-cov/lib/python2.7/site-packages/redis/connection.py", line 619, in send_command
2019-01-24T11:28:53.5379370Z     self.send_packed_command(self.pack_command(*args))
2019-01-24T11:28:53.5380023Z   File "/home/vsts/work/1/s/.tox/py27-cov/lib/python2.7/site-packages/redis/connection.py", line 615, in send_packed_command
2019-01-24T11:28:53.5380372Z     raise e
2019-01-24T11:28:53.5380664Z AttributeError: 'module' object has no attribute 'ensure_binary'
```

**Comment by kapilt** [comment]
new version of fakeredis needs a recent version of six, currently running tests against that.

**Comment by kapilt** [comment]
fixes pushed to #3417
